### PR TITLE
Clean naked sentinels in thoroughly swept tables

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Clean up some trash in a way that doesn't impact the currently running transaction

**Implementation Description (bullets)**:
We use the sweep queue's cache to determine if the table is thoroughly swept and submit the deletes to the delete executor

**Testing (What was existing testing like?  What have you done to improve it?)**:
some integration tests

**Concerns (what feedback would you like?)**:
none

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java [hunk 1]
```diff
@@ -1302,16 +1299,33 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
...
         }
 
+        deleteOrphanedSentinelsAsync(table, actualOrphanedSentinels);
+
         return actualOrphanedSentinels;
     }
 
+    private void deleteOrphanedSentinelsAsync(TableReference table, Set<Cell> actualOrphanedSentinels) {
+        sweepQueue.getSweepStrategy(table).ifPresent(strategy -> {
+            if (strategy == SweeperStrategy.THOROUGH) {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java:1309]: note: it's important that THOROUGH_MIGRATION _does not_ choose the THOROUGH sweep strategy (which it doesn't)

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java [hunk 2]
```diff
@@ -1617,16 +1607,98 @@ public void getSentinelValuesStressTest() {
         assertThatCode(() -> retryTxn.get(TABLE_SWEPT_CONSERVATIVE, cells)).doesNotThrowAnyException();
     }
 
-    private void putUncommittedAtFreshTimestamp(Cell cell) {
+    @Test
+    public void orphanedSentinelsGetDeletedAfterDetectionInThoroughlySweptTable() {
         keyValueService.put(
-                TABLE_SWEPT_CONSERVATIVE,
+                TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, new byte[0]), Value.INVALID_VALUE_TIMESTAMP);
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java:1613]: nit: `writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, TEST_CELL)`?

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java [hunk 3]
```diff
@@ -1617,16 +1607,98 @@ public void getSentinelValuesStressTest() {
...
+        putUncommittedAtFreshTimestamp(TABLE_SWEPT_THOROUGH, TEST_CELL_2);
+
+        assertThatThrownBy(() -> t1.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)))
+                .isInstanceOf(TransactionFailedRetriableException.class)
+                .hasMessageContaining("Tried to read a value that has been deleted.");
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .containsExactlyEntriesOf(
+                        ImmutableMap.of(TEST_CELL, Value.create(new byte[0], Value.INVALID_VALUE_TIMESTAMP)));
+    }
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java:1674]: Did we want to do something about test cells 2 and 3 after writing them? I think you want to read everything on line 1667, and then check the others are still there

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java [hunk 4]
```diff
@@ -1617,16 +1607,98 @@ public void getSentinelValuesStressTest() {
...
 
-    private void writeSentinelToTestTable(Cell cell) {
-        keyValueService.put(
-                TABLE_SWEPT_CONSERVATIVE, ImmutableMap.of(cell, new byte[0]), Value.INVALID_VALUE_TIMESTAMP);
+    private void writeSentinelToTestTable(TableReference tableRef, Cell cell) {
+        keyValueService.put(tableRef, ImmutableMap.of(cell, new byte[0]), Value.INVALID_VALUE_TIMESTAMP);
+    }
+
+    private void commitWrite(TableReference tableSweptThorough, Cell testCell) {
+        Transaction t2 = txManager.createNewTransaction();
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java:1699]: nit: let's not name this `t2`, this came from refactoring

**Review by jeremyk-91 — APPROVED** [review]
👍  we paired on this, broadly looks good. Some small questions around the tests.

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java (hunk 3 above)
Comment (gmaretic) [reply atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java:1674]: oops

**Pull request merged**